core/src/components/range/range.tsx

```diff
@@ -418,42 +424,101 @@ export class Range implements ComponentInterface {
+   */
+  private onStart() {
+    this.ionKnobMoveStart.emit({ value: this.ensureValueInBounds(this.value) });
+  }
 
-    const rect = (this.rect = this.rangeSlider!.getBoundingClientRect() as any);
+  /**
+   * The value should be updated while dragging the
+   * bar or knob.
+   *
+   * While the user is dragging, the view
+   * should not scroll. This is to prevent the user from
+   * feeling disoriented while dragging.
+   *
+   * The user can scroll on the view if the knob or
+   * bar is not being dragged.
+   *
+   * @param detail The details of the gesture event.
+   */
+  private onMove(detail: GestureDetail) {
+    const { contentEl, pressedKnob } = this;
     const currentX = detail.currentX;
 
-    // figure out which knob they started closer to
-    let ratio = clamp(0, (currentX - rect.left) / rect.width, 1);
-    if (isRTL(this.el)) {
-      ratio = 1 - ratio;
+    /**
+     * Since the user is dragging on the bar or knob, the view should not scroll.
+     *
+     * This only needs to be done once.
+     */
+    if (contentEl && this.initialContentScrollY === undefined) {
+      this.initialContentScrollY = disableContentScrollY(contentEl);
     }
 
-    this.pressedKnob = !this.dualKnobs || Math.abs(this.ratioA - ratio) < Math.abs(this.ratioB - ratio) ? 'A' : 'B';
-
-    this.setFocus(this.pressedKnob);
+    /**
+     * The `pressedKnob` can be undefined if the user just
+     * started dragging the knob.
+     *
+     * This is necessary to determine which knob the user is dragging,
+     * especially when it's a dual knob.
+     * Plus, it determines when to apply certain styles.
+     *
+     * This only needs to be done once since the knob won't change
+     * while the user is dragging.
+     */
+    if (pressedKnob === undefined) {
+      this.setPressedKnob(currentX);
+    }
 
-    // update the active knob's position
     this.update(currentX);
-
-    this.ionKnobMoveStart.emit({ value: this.ensureValueInBounds(this.value) });
   }
 
-  private onMove(detail: GestureDetail) {
-    this.update(detail.currentX);
-  }
-
-  private onEnd(detail: GestureDetail) {
+  /**
+   * The value should be updated on touch end:
+   * - When the user lifts their finger from the screen after
+   * tapping the bar.
+   *
+   * @param detail The details of the gesture or mouse event.
+   */
+  private onEnd(detail: GestureDetail | MouseEvent) {
     const { contentEl, initialContentScrollY } = this;
-    if (contentEl) {
+    const currentX = (detail as GestureDetail).currentX || (detail as MouseEvent).clientX;
+
+    /**
+     * The `pressedKnob` can be undefined if the user never
+     * dragged the knob. They just tapped on the bar.
+     *
+     * This is necessary to determine which knob the user is changing,
+     * especially when it's a dual knob.
+     * Plus, it determines when to apply certain styles.
+     */
+    if (this.pressedKnob === undefined) {
+      this.setPressedKnob(currentX);
+    }
```

Comment:
I'm no longer seeing the `ionKnobMoveStart` event emit when clicking on an empty part of the range to move the knob (without dragging), only `ionKnobMoveEnd`. In `main`, both events emit when clicking (first start, then end).

We might be able to emit the start event here? I assume it's not firing since the gesture never technically started.

Reply:
I'm now seeing the events both fire twice total on release of a drag gesture:

https://github.com/ionic-team/ionic-framework/assets/90629384/04318f35-2b81-4206-8a75-fcb580cf4f9e

Reply:
Again good catch! https://github.com/ionic-team/ionic-framework/pull/29005/commits/8fe6d79b6fa5eff5823ec24b2ee1c09e44b82c0c

Reply:
@amandaejohnston ready for review

Reply:
Looks good now, thank you!